_unittest_ironpython/test_00EDB.py

```diff
@@ -17,6 +17,7 @@ def setUpClass(cls):
     @classmethod
     def tearDownClass(cls):
         test_obj.teardown_class()
+        time.sleep(2)
```

Comment:
In any gui testing I've had, setting a sleep to make sure something is finished tends to be brittle and machine/environment dependent.  Is there a way you could check if something is cleaned up and then proceed when it is?  Enter a while loop (with a maximum timeout) and check if something is collected?  For example, you can:

```py
import weakref

obj_ref = weakref.ref()
test_obj.teardown_class()
del test_obj
timeout = time.time() + 2
while obj_ref() is not None:
    time.sleep(0.1)
    if time.time() > timeout:
        break
        # alternatively:
        # raise TimeoutError('Exceeded teardown timeout')
```

Reply:
some methods in AEDT returns True even if they are still running. this may happen for instance in the file save and close that is part of TearDown. I could think to a while loop strategy but I need to do extensive local tests to verify the approach works well.